static/js/components/UploadResumeView.js

```diff
@@ -0,0 +1,153 @@
+// @flow
+/* global SETTINGS: false */
+import React from "react"
+import Dropzone from "react-dropzone-uploader"
+import { compose } from "redux"
+import { connect } from "react-redux"
+import { mutateAsync } from "redux-query"
+
+import applications from "../lib/queries/applications"
+import type { ApplicationDetail } from "../flow/applicationTypes"
+import type { ApplicationResumeDetail } from "../flow/applicationTypes"
+import { ErrorMessage, Field, Form, Formik } from "formik"
+import FormError from "./forms/elements/FormError"
+import { DEFAULT_POST_OPTIONS } from "../lib/redux_query"
+
+const inputContent = (
+  <div className="input-label">
+    <h3>Drag file here or click to Browse</h3>
+    <p>Allowed file types: pdf.</p>
+  </div>
+)
+
+const ResumeUploader = props => {
+  const getUploadParams = () => {
+    return {
+      url: `/api/applications/${props.application.id}/resume/`,
+      ...DEFAULT_POST_OPTIONS
+    }
+  }
+
+  const handleChangeStatus = ({ meta, file }, status) => {
+    console.log(status, meta, file)
+  }
+
+  // receives array of files that are done uploading when submit button is clicked
+  const handleSubmit = (files, allFiles) => {
+    allFiles.forEach(f => f.remove())
+  }
+
+  return (
+    <Dropzone
+      getUploadParams={getUploadParams}
+      onChangeStatus={handleChangeStatus}
+      inputContent={inputContent}
+      classNames="drop-outline"
+      onSubmit={handleSubmit}
+      maxFiles={1}
+      styles={{ dropzone: { minHeight: 200, maxHeight: 250 } }}
+      accept="application/pdf,application/msword"
+    />
+  )
+}
+type Props = {
+  application: ?ApplicationDetail,
+  uploadResume: (
+    applicationDetail: ApplicationResumeDetail,
+    application_id: number
+  ) => Promise<void>
+}
+
+export class UploadResumeView extends React.Component<Props> {
+  render() {
+    const { uploadResume, application } = this.props
+
+    return (
+      <div className="container">
+        <div className="drawer-wrapper">
+          <h2 className="drawer-title">Upload Resume</h2>
+          <p>Please upload resume or a link to your linkedin profile</p>
+
+          <ResumeUploader application={application} />
+          {application.resume_filename ? (
+            <p className="uploaded-resume">
+              You have uploaded your resume: {application.resume_filename}
+            </p>
+          ) : null}
+
+          <Formik
+            onSubmit={async (
+              { linkedin_url, file },
+              { setSubmitting, setErrors }: Object
+            ) => {
+              const {
+                body: errors
+              }: // $FlowFixMe
+              { body: Object } = await uploadResume(
+                {
+                  resume_filename: file,
+                  linkedin_url:    linkedin_url
+                },
+                application.id
+              )
+
+              if (errors && errors.length > 0) {
+                setErrors({
+                  linkedin_url: errors
```

Comment:
Not seeing error messages when something goes wrong (for example when the application is in the wrong state and the response returned is `{errors: "Cannot upload a resume in this application state"}`.  Would be good to also have a notification when the submission is successful.  [I used bootstrap alerts for this in another PR](https://github.com/mitodl/bootcamp-ecommerce/pull/620/files#diff-de0f43dd282cf73824d967d8e2bb1fe5R167-R177).

Reply:
@annagav @mbertrand - We didn't have error/success behavior defined for us initially. Abdul provided feedback on how this should generally work right here: https://github.com/mitodl/bootcamp-ecommerce/issues/641. That doesn't cover every single case where an error might occur from an HTTP request or we have to alert the user about a successful request, but it provides some guidelines. As indicated in that issue, in the case of a success message, we want to bring up a Toast. I think it's fine to completely ignore the error/success behavior in this PR, and instead we can implement it all at once as part of this issue (which I'm going to fill in details for): https://github.com/mitodl/bootcamp-ecommerce/issues/641. We're at risk here of having a whole bunch of custom implementations of error/success messages since we didn't have that behavior spec'ed out initially. It might be better just to ignore it for now and knock it all out in one PR, but I'll leave it to you @annagav to decide if some of that behavior is simple enough that it makes sense to include in this PR